src/Filters/LinearKalmanFilter.h

```diff
@@ -0,0 +1,46 @@
+#ifndef LINEARKALMANFILTER_H
+#define LINEARKALMANFILTER_H
+
+#include "Filter.h"
+#include "../Math/Matrix.h"
+
+namespace mmfs {
+
+class LinearKalmanFilter : public Filter {
+protected:
+    struct State {
+        Matrix X; // State vector
+        Matrix U; // Control input
+        Matrix P; // Covariance matrix
+        Matrix F; // State transition matrix
+        Matrix G; // Control input model
+        Matrix Q; // Process noise covariance
+        Matrix R; // Measurement noise covariance
+        Matrix H; // Measurement model
+        Matrix K; // Kalman gain
+    } state;
+
+    // Customizable methods for subclasses
+    virtual Matrix setF(double dt) = 0;
+    virtual Matrix setG(double dt) = 0;
+    virtual Matrix setH() = 0;
+    virtual void calculate_initial_values() = 0;
+
+public:
+    LinearKalmanFilter::LinearKalmanFilter(Matrix X, Matrix U, Matrix P, Matrix F, Matrix G, Matrix R, Matrix Q);
```

Comment:
We should add a comment that the X, U and P passed in are the initials values while the other 4 matrices are constant